bitcoin/rpc.py

```diff
@@ -449,11 +383,12 @@ def getblockheader(self, block_hash, verbose=False):
 
         Raises IndexError if block_hash is not valid.
         """
-        try:
-            block_hash = b2lx(block_hash)
-        except TypeError:
-            raise TypeError('%s.getblockheader(): block_hash must be bytes; got %r instance' %
-                    (self.__class__.__name__, block_hash.__class__))
+        if type(block_hash) != str:
```

Comment:
I think it would be better to use `if not isinstance(block_hash, str):`

Reply:
Could you explain the benefit? I used isinstance for any CObject or bytes (some of which could be subclassed), but I assumed str would never be subclassed, so I used type when checking against str. Is this not the right approach?

Reply:
I think that the library should not make assumptions about what the calling code could do or not do. Or if it makes assumptions, it should state them explicitly, and produce error if assumptions are violated.

Also, the caller might not even know if they are working with subclass of `str`. They could received the value from some other library that decided to subclass `str` for some internal reason.

Reply:
I just committed again, changing all type() to isinstance(). Thanks for the advice!